Userland/Libraries/LibJS/AST.cpp

```diff
@@ -3227,41 +3228,130 @@ void ImportCall::dump(int indent) const
+// Also includes assertions from proposal: https://tc39.es/proposal-import-assertions/#sec-import-call-runtime-semantics-evaluation
 Completion ImportCall::execute(Interpreter& interpreter, GlobalObject& global_object) const
 {
     InterpreterNodeScope node_scope { interpreter, *this };
-    // 1. Let referencingScriptOrModule be ! GetActiveScriptOrModule().
-    auto referencing_script_or_module = interpreter.vm().get_active_script_or_module();
 
-    if (m_options)
-        return interpreter.vm().throw_completion<InternalError>(global_object, ErrorType::NotImplemented, "import call with assertions/options");
+    // 2.1.1.1 EvaluateImportCall ( specifierExpression [ , optionsExpression ] ), https://tc39.es/proposal-import-assertions/#sec-evaluate-import-call
+    //  1. Let referencingScriptOrModule be ! GetActiveScriptOrModule().
+    auto referencing_script_or_module = interpreter.vm().get_active_script_or_module();
 
-    // 2. Let argRef be the result of evaluating AssignmentExpression.
-    // 3. Let specifier be ? GetValue(argRef).
+    // 2. Let specifierRef be the result of evaluating specifierExpression.
+    // 3. Let specifier be ? GetValue(specifierRef).
     auto specifier = TRY(m_specifier->execute(interpreter, global_object));
 
-    // 4. Let promiseCapability be ! NewPromiseCapability(%Promise%).
+    auto optionsValue = js_undefined();
+    // 4. If optionsExpression is present, then
+    if (m_options) {
+        // a. Let optionsRef be the result of evaluating optionsExpression.
+        // b. Let options be ? GetValue(optionsRef).
+        optionsValue = TRY(m_options->execute(interpreter, global_object)).release_value();
+    }
+    // 5. Else,
+    // a. Let options be undefined.
+    // Note: options_value is undefined by default.
+
+    // 6. Let promiseCapability be ! NewPromiseCapability(%Promise%).
     auto promise_capability = MUST(new_promise_capability(global_object, global_object.promise_constructor()));
 
-    VERIFY(!interpreter.exception());
-    // 5. Let specifierString be ToString(specifier).
-    auto specifier_string = specifier->to_string(global_object);
-
-    // 6. IfAbruptRejectPromise(specifierString, promiseCapability).
-    // Note: Since we have to use completions and not ThrowCompletionOr's in AST we have to do this manually.
-    if (specifier_string.is_throw_completion()) {
-        // FIXME: We shouldn't have to clear this exception
-        interpreter.vm().clear_exception();
-        (void)TRY(call(global_object, promise_capability.reject, js_undefined(), *specifier_string.throw_completion().value()));
-        return Value { promise_capability.promise };
+    // 27.2.1.1.1 IfAbruptRejectPromise ( value, capability ), https://tc39.es/ecma262/#sec-ifabruptrejectpromise
+#define TRY_OR_REJECT_WITH_COMPLETION(global_object, capability, expression)                                                            \
```

Comment:
The name doesn't really seem accurate? This just returns a value. Please also move it next to the other one instead of here where it can't be reused.